mergify_engine/debug.py

```diff
@@ -152,28 +153,37 @@ async def _get_config(
         print(f"** {formatted_pulls} (priority: {fancy_priority})")
 
 
-async def report(
+def _parser(
     url: str,
-) -> typing.Union[context.Context, github.AsyncGithubInstallationClient, None]:
-    redis_cache = utils.create_aredis_for_cache(max_idle_time=0)
+) -> typing.Tuple[github_types.GitHubLogin, typing.Optional[str], typing.Optional[str]]:
 
-    path = url.replace("https://github.com/", "")
+    path = urllib.parse.urlparse(url).path.split("/")
 
     pull_number: typing.Optional[str]
     repo: typing.Optional[str]
 
     try:
-        owner, repo, _, pull_number = path.split("/")
+        owner, repo, _, pull_number = [el for el in path if el != ""]
     except ValueError:
         pull_number = None
         try:
-            owner, repo = path.split("/")
+            _, owner, repo = path
         except ValueError:
-            owner = path
+            _, owner = path
             repo = None
 
     owner = typing.cast(github_types.GitHubLogin, owner)
 
+    return owner, repo, pull_number
+
+
+async def report(
+    url: str,
+) -> typing.Union[context.Context, github.AsyncGithubInstallationClient, None]:
+    redis_cache = utils.create_aredis_for_cache(max_idle_time=0)
+
+    owner, repo, pull_number = _parser(url)
```

Comment:
Since `_parser()` can raise `ValueError`, it should be caught here, to print error message and exit.

Reply:
Maybe _parser can have a better name